openage/convert/dataformat/format_metadata.py

```diff
@@ -0,0 +1,154 @@
+
+    def add_layer(self, layer_id, mode, position=None, time_per_frame=None, replay_delay=None):
+        layer_def = ['layer {} mode={}'.format(layer_id, mode)]
+        if position is not None:
+            layer_def.append(' position={}'.format(position))
+        if time_per_frame is not None:
+            layer_def.append(' time_per_frame={}'.format(time_per_frame))
+        if replay_delay is not None:
+            layer_def.append(' replay_delay={}'.format(replay_delay))
+        layer_def.append('\n')
+        out = ''.join(layer_def)
+        self.txt.append(out)
+
+    def add_frame(self, *args):
+        frame_def = ['frame']
+        frame_def.append(' '.join(args))
+        frame_def.append('\n')
+        out = ''.join(frame_def)
+        self.txt.append(out)
+
+    def format_data(self):
+        pass
+
+    def dump(self):
+        return ''.join(self.txt)
+
+    def __str__(self):
+        return 'data type: {}\ndata name: {}'.format(self.type, self.name)
+
+
+class SpriteDefinition(DataDefinition):
+    type = 'sprite'
+    version = 0
+
+    def add_angle(self, degree, mirror_from=None):
+        angle_def = ['angle {}'.format(degree)]
+        if mirror_from is not None:
+            angle_def.append(' mirror_from={}'.format(mirror_from))
+        angle_def.append('\n')
+        out = ''.join(angle_def)
+        self.txt.append(out)
+
+    def format_data(self):
+        self.add_header()
+        self.add_version()
+        for image_id, file in self.data['image_files'].items():
+            self.add_image(image_id, file)
+        for layer_id, kwargs in self.data['layers'].items():
+            self.add_layer(layer_id, **kwargs)
+        for degree, args in self.data['angles'].items():
+            self.add_angle(degree, args['mirror_from'])
+            for frame in args['frames']:
+                self.add_frame(*frame)
+
+
+class TerrainDefinition(DataDefinition):
+    type = 'terrain'
+    version = 0
+
+    def add_blending_priority(self, priority):
+        self.txt.append('blending_priority {}\n'.format(priority))
+
+    def add_blending_mask(self, blend_id, filename):
+        self.txt.append('blending_mask {} {}\n'.format(blend_id, filename))
+
+    def add_dots_per_tile(self, dots_number):
+        self.txt.append('dots_per_tile {}\n'.format(dots_number))
+
+    def format_data(self):
+        self.add_header()
+        self.add_version()
+        self.add_blending_priority(self.data['blending_priority'])
+        self.add_blending_mask(**self.data['blending_mask'])
+        self.add_dots_per_tile(self.data['dots_per_tile'])
+        for image_id, file in self.data['image_files'].items():
+            self.add_image(image_id, file)
+        for layer_id, kwargs in self.data['layers'].items():
+            self.add_layer(layer_id, **kwargs)
+        for frame in self.data['frames']:
+            self.add_frame(*frame)
+
+
+if __name__ == '__main__':
+    dummy_sprite_data = {
```

Comment:
If you want to test the functionality of your code, you should use the methods that you created rather than another data format. The methods will be what we call in the converter after all :D You can import the module into the python console and call the methods there.